akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -116,7 +117,39 @@ private[akka] trait InboundContext {
  */
 private[akka] object AssociationState {
   def apply(): AssociationState =
-    new AssociationState(incarnation = 1, uniqueRemoteAddressPromise = Promise(), quarantined = Set.empty)
+    new AssociationState(incarnation = 1, uniqueRemoteAddressPromise = Promise(), quarantined = QuarantinedUidSet.empty)
+
+  object QuarantinedUidSet {
```

Comment:
Is this written so often that this deserves such an optimization? I like that it is array backed though since that is faster to traverse, but I am not sure this is worth it.